### PR TITLE
[#ENTE-79] new UpdateWebviewServicesMetadata

### PR DESCRIPTION
**Pull request by BurnedMarshal**
#### List of Changes

Create new UpdateWebviewServicesMetadata that read all the services from cosmos and export two file into the storage with the json needed from the website to generate the services webpages and webview.

#### Motivation and Context

This operation is executed manually every day, we automate the process with this time triggered activity.

#### How Has This Been Tested?

Unit Test

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: UpdateWebviewServicesMetadata/handler.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  }, {} as Record<string, ServicesExportCompact | ServicesExportExtended>);
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const UpdateWebviewServicesMetadata = (
+  serviceModel: ServiceModel,
+  serviceIdExclusionList: ReadonlyArray<NonEmptyString>
+) => async (context: Context): Promise<unknown> =>
+  pipe(
+    serviceModel.listLastVersionServices(),
+    TE.mapLeft(comsosError => new Error(`CosmosError: ${comsosError.kind}`)),
```
Comment (AleDore) [inline UpdateWebviewServicesMetadata/handler.ts:154]: typo

File: utils/config.ts [hunk 2]
```diff
@@ -24,8 +26,14 @@ export const IConfig = t.interface({
   AssetsStorageConnection: NonEmptyString,
   AzureWebJobsStorage: NonEmptyString,
 
+  COSMOSDB_KEY: NonEmptyString,
+  COSMOSDB_NAME: NonEmptyString,
+  COSMOSDB_URI: NonEmptyString,
+
   MaxServicesOrchestratorSize: NonNegativeInteger,
 
+  SERVICEID_EXCLUSION_LIST: t.readonlyArray(NonEmptyString),
```
Comment (AleDore) [inline utils/config.ts:35]: Can we use `CommaSeparatedArrayOf`?

File: UpdateWebviewServicesMetadata/handler.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+import * as t from "io-ts";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as O from "fp-ts/lib/Option";
+import * as T from "fp-ts/lib/Task";
+
+const ServiceExportCompact = t.interface({
+  i: NonEmptyString,
+  n: NonEmptyString,
+  q: t.number
```
Comment (AleDore) [inline UpdateWebviewServicesMetadata/handler.ts:24]: Can't we rename these with more meaningful ones?

Comment (BurnedMarshal) [reply UpdateWebviewServicesMetadata/handler.ts:24]: Is the purpose of this JSON prop to be minimized to reduce the size of the output JSON consumed by the website.
I can add some comments into the code, but the prop name must be 1 character long.

File: UpdateWebviewServicesMetadata/handler.ts (hunk 1 above)
Comment (BurnedMarshal) [reply UpdateWebviewServicesMetadata/handler.ts]: fixed in cc7c170

File: utils/config.ts (hunk 2 above)
Comment (BurnedMarshal) [reply utils/config.ts]: Fixed in 1199072

File: UpdateWebviewServicesMetadata/handler.ts (hunk 3 above)
Comment (BurnedMarshal) [reply UpdateWebviewServicesMetadata/handler.ts]: Fixed in 601fbdf

**Review by AleDore — APPROVED** [review]
LGTM

**Pull request merged**